x/mongo/driver/mongocrypt/mongocrypt.go

```diff
@@ -181,8 +186,35 @@ func (m *MongoCrypt) CreateExplicitEncryptionContext(doc bsoncore.Document, opts
 
 	algoStr := C.CString(opts.Algorithm)
 	defer C.free(unsafe.Pointer(algoStr))
-	if ok := C.mongocrypt_ctx_setopt_algorithm(ctx.wrapped, algoStr, -1); !ok {
-		return nil, ctx.createErrorFromStatus()
+
+	if opts.Algorithm == "Indexed" {
+		if ok := C.mongocrypt_ctx_setopt_index_type(ctx.wrapped, IndexTypeIndexed); !ok {
+			return nil, ctx.createErrorFromStatus()
+		}
+	} else if opts.Algorithm == "Unindexed" {
```

Comment:
Optional: Consider turning this into a `switch` statement instead to make it more readable.

E.g.
```go
switch opts.Algorithm {
case "Indexed":
	// ...
case "Unindexed":
	// ...
default:
	// ...
}

Reply:
Done.